### ISSUE TITLE
Have proper versioning of e2e npm package

### ISSUE THREAD

**Issue by rhopp** [eclipse-che/che#18771]
### Is your task related to a problem? Please describe.
We would like to start publishing orh typescript e2e tests package on npm registry for easier dependency management in downstream.

### Describe the solution you'd like
- [x]  Rename e2e package to che-e2e (or any other clearer name)
- [x] Package version should follow Che version
- [x] Upversioning should be hooked up into release process pipelines

### Additional info
Once this is done, we can start working on proper publishing the package into npm registry as part of the nightly/release pipelines. This will be done in followup issue: Publish che-e2e npm package into public registry #18772

Story points: 5

**Comment by nickboldt** [comment]
Related task: moving the test code from `eclipse/che` repo to its own repo, `eclipse-che/che-tests`

See https://issues.redhat.com/browse/CRW-1732

**Comment by nickboldt** [comment]
Work started this past week on this issue but it is unclear how much progress has been made, as Mykhailo did not provide a standup / status before going on PTO for two weeks.

**Comment by l0rd** [comment]
Moving to milestone 7.40 so that we can close milestone 7.39

**Comment by mkuznyetsov** [comment]
PR to cover remaining items in checklist https://github.com/eclipse/che/pull/20846

**Comment by nickboldt** [comment]
PR merged. So I guess we run this on Wed and see what happens :)
